go/adbc/driver/snowflake: integration tests are failing

```
[ RUN      ] SnowflakeStatementTest.SqlIngestTimestamp
panic: interface conversion: arrow.Array is *array.Int64, not *array.Timestamp

goroutine 455 [running]:
github.com/apache/arrow-adbc/go/adbc/driver/snowflake.getTransformer.func6({0x7f11d57bc318?, 0x10c000610640?}, {0x7f11d57dd400, 0x10c0006544c0?})
	/home/runner/work/arrow-adbc/arrow-adbc/go/adbc/driver/snowflake/record_reader.go:169 +0x5c5
github.com/apache/arrow-adbc/go/adbc/driver/snowflake.getRecTransformer.func1({0x7f11d57bc318, 0x10c000610640}, {0x7f11d57da070, 0x10c0008e56e0})
	/home/runner/work/arrow-adbc/arrow-adbc/go/adbc/driver/snowflake/record_reader.go:59 +0x255
github.com/apache/arrow-adbc/go/adbc/driver/snowflake.newRecordReader.func2()
	/home/runner/work/arrow-adbc/arrow-adbc/go/adbc/driver/snowflake/record_reader.go:302 +0x1c3
golang.org/x/sync/errgroup.(*Group).Go.func1()
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.3.0/errgroup/errgroup.go:75 +0x64
created by golang.org/x/sync/errgroup.(*Group).Go
	/home/runner/go/pkg/mod/golang.org/x/sync@v0.3.0/errgroup/errgroup.go:72 +0xa5
```

--- a/go/adbc/driver/snowflake/record_reader.go
+++ b/go/adbc/driver/snowflake/record_reader.go
@@ -133,7 +133,7 @@ func getTransformer(sc *arrow.Schema, ld gosnowflake.ArrowStreamLoader) (*arrow.
 						tb.Append(arrow.Timestamp(time.Unix(epoch[i], int64(fraction[i])).UnixNano()))
 					}
 				} else {
-					for i, t := range a.(*array.Timestamp).TimestampValues() {
+					for i, t := range a.(*array.Int64).Int64Values() {
 						if a.IsNull(i) {
 							tb.AppendNull()
 							continue
@@ -166,7 +166,7 @@ func getTransformer(sc *arrow.Schema, ld gosnowflake.ArrowStreamLoader) (*arrow.
 						tb.Append(arrow.Timestamp(time.Unix(epoch[i], int64(fraction[i])).UnixNano()))
 					}
 				} else {
-					for i, t := range a.(*array.Timestamp).TimestampValues() {
+					for i, t := range a.(*array.Int64).Int64Values() {
 						if a.IsNull(i) {
 							tb.AppendNull()
 							continue

